### DISCUSSION TITLE
加入简单的 progressive web application ( PWA )

### DISCUSSION THREAD

**Discussion by NoverNobida** [usememos/memos#133] [Ideas]
PWA可以在电脑，安卓或者ios上作为软件安装，使体验增强。复杂的PWA可以使用API之类的，可以快速添加笔记。
之前都是直接编辑替换docker镜像里面的 `/usr/local/memos/web/dist` 目录，但是目前项目似乎将前端文件与主程序整合到一起了，想要修改只能自行编译。另外，编辑替换 `/usr/local/memos/web/dist` 目录的另一个原因是网站图标 `favicon.ico` 实际上是往右偏的，在浏览器水平标签页里面不容易看出，但在垂直标签页十分明显。
![图片](https://user-images.githubusercontent.com/46912250/180960267-a01fbeb8-3b46-4d4f-a83c-df71b2138632.png)

**Comment by boojack** [comment]
这个 icon 是一个 emoji，各个系统会显示不一样，之后会把它换成静态的 iOS 上的图片。👇
![memos-logo](https://user-images.githubusercontent.com/24653555/180970906-dc15e4fd-6b5c-4997-8032-fa637ef26121.png)

**Reply by boojack** [reply]
Updated in https://github.com/usememos/memos/commit/873973a0888500bfdcfdc46cf2838cda3b1494e6

**Comment by boojack** [comment]
至于 PWA，我想只要在 `./web/public/` 加一个 `manifest.json` 就可以了，或许你可以提个 PR
